Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -3874,4 +3898,62 @@ public void addExtraInstrumentationData(HashMap<String, String> instrumentationD
     public void addExtraInstrumentationData(String key, String value) {
         instrumentationExtraData_.put(key, value);
     }
+
+
+    //-------------------- Promo view handling--------------------//
+
+
+    @Override
+    public void onPromoViewVisible(String action) {
+        //No Implementation on purpose
+    }
+
+    @Override
+    public void onPromoViewDismissed(String action) {
+        if(ServerRequestInitSession.isInitSessionAction(action)){
+            checkForAutoDeepLinkConfiguration();
+        }
+    }
+
+    @Override
+    public void onPromoViewError(int errorCode, String errorMsg) {
+
+    }
+
+    /**
+     * Interface for defining optional promo view behaviour for Activities
+     */
+    public interface IPromoViewControl{
+        /**
+         * Defines if an activity is interested to show promo views or not.
+         * By default activities are considered as promo view enabled. In case of activities which are not interested to show a promo view (Splash screen for example)
+         * should implement this and return false. The pending promo view will be shown with the very next promo view activity
+         *
+         * @return A {@link Boolean} whose value is true if the activity don't want to show any promo view.
+         */
+        boolean skipPromoViewsOnThisActivity();
+    }
+
+    //--------- Test data to simulate Promo view------------//
+    //TODO should be part of initSession JsonObj. Delete as back end implement this feature.
+    public JSONArray getPromoViewData() {
+        JSONArray promoViewArray  = null;
+        String previewArrayObj = "{\"app_promo_data\":[{"
+                + " \"app_promo_id\":\"promo_id_01\","
+                + " \"app_promo_action\" : \"open\","
+                + " \"num_of_use\":1,"
+                + " \"promo_view_url\":\"https://branch.io\","
+                + " \"expiry\":123456778 },{"
+                + " \"app_promo_id\":\"promo_id_02\","
+                + " \"app_promo_action\" : \"buy\","
+                + "  \"num_of_use\":1,"
+                + " \"promo_view_url\":\"https://branch.io\","
```

Comment:
Do you think we should use html text instead of url? Since this is received
aling with open/install I though of not overloading them much
On Mar 8, 2016 10:32 PM, "Alex Austin" <notifications@github.com> wrote:

> In Branch-SDK/src/io/branch/referral/Branch.java
> <https://github.com/BranchMetrics/Android-Deferred-Deep-Linking-SDK/pull/220#discussion_r55477309>
> :
>
> > +    }
> > +
> > +    //--------- Test data to simulate Promo view------------//
> > +    //TODO should be part of initSession JsonObj. Delete as back end implement this feature.
> > +    public JSONArray getPromoViewData() {
> > +        JSONArray promoViewArray  = null;
> > +        String previewArrayObj = "{\"app_promo_data\":[{"
> > +                + " \"app_promo_id\":\"promo_id_01\","
> > +                + " \"app_promo_action\" : \"open\","
> > +                + " \"num_of_use\":1,"
> > +                + " \"promo_view_url\":\"https://branch.io\","
> > +                + " \"expiry\":123456778 },{"
> > +                + " \"app_promo_id\":\"promo_id_02\","
> > +                + " \"app_promo_action\" : \"buy\","
> > +                + "  \"num_of_use\":1,"
> > +                + " \"promo_view_url\":\"https://branch.io\","
>
> Can you implement an HTML variable as the preference to URL? Similar to
> what we did with BX.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/BranchMetrics/Android-Deferred-Deep-Linking-SDK/pull/220/files#r55477309>
> .
>